### PR TITLE
Archive breadcrumb with new api

### LINKED ISSUE #5414: Make the archive breadcrumb work with the new API

### PR DESCRIPTION
**Pull request by gestchild**
Fixes #5414

### CODE REVIEW

File: catalogue/webapp/services/catalogue/works.js [hunk 1]
```diff
@@ -79,7 +79,9 @@ export async function getWork({
   const params = {
     include: [
       ...workIncludes,
-      ...(toggles && toggles.archivesPrototype ? ['collection'] : []),
+      ...(toggles && toggles.archivesPrototype
```
Comment (jamesgorrie) [inline catalogue/webapp/services/catalogue/works.js:82]: Can we not just put this in?

File: common/utils/works.js [hunk 2]
```diff
@@ -234,6 +234,14 @@ export function getItemIdentifiersWith(
   }, []);
 }
 
+export type ArchiveNode = {|
+  id: string,
+  title: string,
+  alternativeTitles: string[],
+  referenceNumber: string,
+  type: 'Work',
+|};
```
Comment (jamesgorrie) [inline common/utils/works.js:243]: Could we start porting this to TS?

File: common/views/components/ArchiveBreadcrumb/ArchiveBreadcrumb.js [hunk 3]
```diff
@@ -81,100 +78,53 @@ type Props = {|
...
-      setBreadcrumb(makeCrumbs(tree));
-    }
-  }, [work]);
-
-  const [breadcrumb, setBreadcrumb] = useState({
-    firstCrumb: null,
-    middleCrumbs: [],
-    lastCrumb: null,
-  });
+  const firstCrumb = work.partOf[0];
```
Comment (jamesgorrie) [inline common/views/components/ArchiveBreadcrumb/ArchiveBreadcrumb.js:82]: Is this guaranteed to exist?

Comment (jamesgorrie) [reply common/views/components/ArchiveBreadcrumb/ArchiveBreadcrumb.js:82]: Ah, looking below, this could be `undefined`.

Comment (jamesgorrie) [reply common/views/components/ArchiveBreadcrumb/ArchiveBreadcrumb.js:82]: Could be worth typing to make this clearer.

File: catalogue/webapp/services/catalogue/works.js (hunk 1 above)
Comment (gestchild) [reply catalogue/webapp/services/catalogue/works.js:82]: If we're happy with the performance of the API with these, then yeah

File: common/utils/works.js (hunk 2 above)
Comment (gestchild) [reply common/utils/works.js:243]: Will do, but in another PR. This is still a prototype, so will make the changes if/when we pursue it, after testing.

File: catalogue/webapp/services/catalogue/works.js (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/services/catalogue/works.js:82]: @warrd would you be happy for this?

Comment (warrd) [reply catalogue/webapp/services/catalogue/works.js:82]: I think we should move as much as possible to the new API if that's what you are asking? I would like to remove the old API code when ready.

Comment (gestchild) [reply catalogue/webapp/services/catalogue/works.js:82]: @warrd it's more about if we're happy for these includes (parts, partOf etc.) to go in by default for all users, in terms of API performance

Comment (warrd) [reply catalogue/webapp/services/catalogue/works.js:82]: I don't know if the new API is slower than the old one, is it that there is more separate requests using this method? I am happy if other people are.

Comment (jamesgorrie) [reply catalogue/webapp/services/catalogue/works.js:82]: 👍

**Pull request merged**